src/main/java/org/amoseman/budgetingbackend/password/ResultType.java

```diff
@@ -0,0 +1,19 @@
+package org.amoseman.budgetingbackend.password;
+
+/**
+ * Represents the type of result from a password strength check.
+ * SUCCESS              - The check is a success.
+ * INVALID_CHARACTER    - The password contains an invalid character, such as a whitespace.
+ * LESS_THAN_MIN_LENGTH - The password is shorter than the configured minimum length.
+ * MISSING_UPPERCASE    - The password is missing an uppercase letter.
+ * MISSING_SPECIAL      - The password is missing a special character.
+ * WEAK_SCORE           - The password has a poor score based on its entropy and bad features, which are repetition and sequences.
+ */
+public enum ResultType {
```

Comment:
The name _ResultType_ makes me think there are multiple types of _Result_s that are possible. Something like _ValidationResult_ or _PasswordTestResult_ maybe possibilities that more clearly state what the enum represents.